src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/NamespaceCache.java

```diff
@@ -41,14 +42,24 @@ protected Uni<CollectionSettings> getCollectionProperties(String collectionName)
           .transformToUni(
               (result, error) -> {
                 if (null != error) {
+                  // collection does not exist
+                  if (error instanceof RuntimeException rte
+                      && rte.getMessage()
+                          .startsWith(ErrorCode.INVALID_COLLECTION_NAME.getMessage())) {
```

Comment:
Not wrong, but in my experience it is usually best practice to check against the error code or some other constant other than the error message, because (1) error messages may change, (2) it is possible that there may be multiple types of errors that start with the same message. Not a blocker, just figured I'd point this out.